lib/publiccloud/provider.pm

```diff
@@ -444,18 +447,25 @@ sub terraform_destroy {
     my $cmd;
     return if get_required_var('TEST') =~ /^publiccloud_upload_img/;
     record_info('INFO', 'Removing terraform plan...');
-    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+    f (get_var('PUBLIC_CLOUD_SLES4SAP')) {
         assert_script_run('cd ' . TERRAFORM_DIR . '/' . $self->conv_openqa_tf_name);
-        $cmd = 'terraform destroy -no-color -auto-approve';
     }
     else {
         assert_script_run('cd ' . TERRAFORM_DIR);
-        # Add region variable also to `terraform destroy` (poo#63604) -- needed by AWS.
-        $cmd = sprintf(q(terraform destroy -no-color -auto-approve -var 'region=%s'), $self->region);
     }
+
+    # Add region variable also to `terraform destroy` (poo#63604) -- needed by AWS.
+    my $region_var  = get_var('PUBLIC_CLOUD_SLES4SAP') ? 'aws_region' : 'region';
+    my $destroy_cmd = sprintf(q(terraform destroy -no-color -auto-approve -var '%s=%s'), $region_var, $self->region);
```

Comment:
Why this extra complexity? Is `terraform destroy` not using `terraform.tfvars` file?

Reply:
What do you mean? It was the code used before, I just changed the `region` variable name.
As far as I understood the `region` variable should be forced on command line because of an issue with the AWS provider. Otherwise I agree with you: the `destroy` command uses the `terraform.tfvars` file.

Reply:
I've tried to simplify.

Reply:
we can go other way around - because HA/SAP can not change their variable names we can align **our** variables to HA/SAP . @cfconrad  WDYT ? ( but it is for sure out of scope for this PR )

Reply:
@asmorodskyi I'm not sure that this is a good idea, simply because the variables are set in another project (ha-sap-terraform-deployments) and can changes (and in fact there are changes in every major versions...) and these changes are not from us.
So maybe better to keep it as is ;-)

Reply:
> What do you mean? It was the code used before, I just changed the `region` variable name.
> As far as I understood the `region` variable should be forced on command line because of an issue with the AWS provider. Otherwise I agree with you: the `destroy` command uses the `terraform.tfvars` file.

From my understanding, the problem on our side occur, because we have such a line: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/publiccloud/terraform/ec2.tf#L19 and with this provider section and an unset variable `region` it fails on destroy.
I just tried our tf file with a `terraform.tfvars` file containing only the region, it worked.

Without knowing your exact tf-file, I think that you have a "hardcoded" `terraform.tfvars` file, you do not need this extra variable.
So I wold like to avoid that extra complexity if it isn't needed. Otherwise nobody will understand it later and we never get rid of it.

@asmorodskyi I think using a `terraform.tfvars` file would help here, the downside is, we cannot just c&p the terraform command to retry it manually. But we could also upload the `tfvars` file as the hasap tests do it.

Reply:
@cfconrad ok, so yes in that case I will revert back to your version and retest to be sure. I thought it was a regression in AWS in all cases.

@ricardobranco777 @asmorodskyi if I revert back to the previous version I will let the `sprintf` as it was already here before ;-)